src/common/image.c

```diff
@@ -1394,7 +1394,8 @@ static uint32_t _image_import_internal(const int32_t film_id, const char *filena
   char *normalized_filename = dt_util_normalize_path(filename);
   if(!normalized_filename
      || !g_file_test(normalized_filename, G_FILE_TEST_IS_REGULAR)
-     || dt_util_get_file_size(normalized_filename) == 0)
+     || dt_util_get_file_size(normalized_filename) == 0
```

Comment:
If you look into the definition of dt_util_get_file_size in utility.c, i think you should compare to <1 instead of ==0

Reply:
This is 3 times calling stat() on the file path. Please reduce it to one call!

Reply:
@cryptomilk you mentioned this twice. file size and regular file can be done with one stat, sure. But access()? isn't there more involved than just bit-mask checking for uid/gid?